packages/swap/src/routes/__tests__/v2/quote.test.ts

```diff
@@ -429,6 +408,96 @@ describe('server', () => {
       expect(sendSpy).toHaveBeenCalledTimes(1);
     });
 
+    it("uses 0 as ingress fee when the protocol can't estimate it", async () => {
+      vi.mocked(getTotalLiquidity).mockResolvedValueOnce(BigInt(2e18));
+      const rpcEnvironment = environment({ maxSwapAmount: null });
+      rpcEnvironment.result.ingress_egress.ingress_fees.Ethereum.USDC = null;
+      mockRpcResponse((url, data: any) => {
+        if (data.method === 'cf_environment') {
+          return Promise.resolve({
+            data: rpcEnvironment,
+          });
+        }
+
+        if (data.method === 'cf_boost_pools_depth') {
+          return Promise.resolve({
+            data: boostPoolsDepth(),
+          });
+        }
+
+        if (data.method === 'cf_pool_depth') {
+          return Promise.resolve({
+            data: cfPoolDepth(),
+          });
+        }
+
+        if (data.method === 'cf_accounts') {
+          return Promise.resolve({
+            data: {
+              id: 1,
+              jsonrpc: '2.0',
+              result: [
+                ['cFMYYJ9F1r1pRo3NBbnQDVRVRwY9tYem39gcfKZddPjvfsFfH', 'Chainflip Testnet Broker 2'],
+              ],
+            },
+          });
+        }
+
+        if (data.method === 'cf_account_info') {
+          return Promise.resolve({
+            data: cfAccountInfo(),
+          });
+        }
+
+        throw new Error(`unexpected axios call to ${url}: ${JSON.stringify(data)}`);
+      });
+      vi.spyOn(WsClient.prototype, 'sendRequest').mockResolvedValueOnce({
+        broker_commission: buildFee('Usdc', 0).bigint,
+        ingress_fee: buildFee('Usdc', 0).bigint,
+        egress_fee: buildFee('Eth', 0).bigint,
+        network_fee: buildFee('Usdc', 100100).bigint,
+        intermediary: null,
+        output: BigInt(1e18),
+      });
+
+      const params = new URLSearchParams({
+        srcChain: 'Ethereum',
+        srcAsset: 'USDC',
+        destChain: 'Ethereum',
+        destAsset: 'ETH',
+        amount: (100e6).toString(),
+      });
+
+      const { body, status } = await request(server).get(`/v2/quote?${params.toString()}`);
+
+      expect(status).toBe(200);
+      expect(body).toMatchObject([
+        {
+          egressAmount: '1000000000000000000',
+          recommendedSlippageTolerancePercent: 1,
+          includedFees: [
+            { type: 'INGRESS', chain: 'Ethereum', asset: 'USDC', amount: '0' },
```

Comment:
might make sense to filter the 0 fees out

Reply:
that affects also egress side so i wasn't sure if i should do it now, or maybe in a follow up